api/firmware/device.go

```diff
@@ -241,11 +258,26 @@ func (device *Device) Init() error {
 	device.channelHashDeviceVerified = false
 	device.sendCipher = nil
 	device.receiveCipher = nil
-	device.changeStatus(StatusConnected)
 
-	if err := device.inferVersionAndProduct(); err != nil {
-		return err
+	if device.version == nil || device.version.AtLeast(semver.NewSemVer(4, 3, 0)) {
```

Comment:
If the version is below 4.3, you can't call it either really as it does not exist, so the 2nd condition alone is enough. 

I don't think you need to call `inferVersionAndProduct` at all really. It helped to avoid the OP_INFO call if the version was already known, but you call it now always, so its not really doing anything. Maybe you could compare the version/product to the one already set if available as a sanity check, but you can just also override it directly.

Reply:
I don't understand.
- we want to call `info()` **always** except if the version is **known** before 4.3. doesn't this mean we call it **either** if version is `nil` **or** if its not `nil` but higher than 4.3 ?
- why is it an error if fw is older than 4.3 and nil ? `device.version.AtLeast()` will be `false` (_older_) if `device.version` is `nil`
>I don't think you need to call inferVersionAndProduct at all really. 

I agree, let me know what you think is best, sanity check, override or just removing the function.

Reply:
1. Thinking about this again, I think removing the `inferVersionAndProduct` makes most sense.
2. Actually without the first condition there would be a nil pointer dereference error if the version is actually `nil`, this crashes:
```
-       if device.version == nil || device.version.AtLeast(semver.NewSemVer(4, 3, 0)) {
+       device.version = nil
+       if device.version.AtLeast(semver.NewSemVer(4, 3, 0)) {
```
I think changing it to `if (device.version == nil || device.version.AtLeast(semver.NewSemVer(9, 2, 0)))` makes sense:
We always want to check `info()` because we're interested in the `initialized` field:
- if we know the version and we know it supports the `initialized` field we want to call `info()`
- if the version is `nil` we want to call `info()` like we did before in `inferVersionAndProduct`
- if we know the version but we know we won't learn the `initialized` field from calling `info()` there is no need to call it anyways, just like `inferVersionAndProduct` avoided calling it if the version was already known.

(unimportant) I think I initially used `AtLeast(4,3,0)` because I thought it makes sense since `info()` will only work for at least this version, but I did not see that if we know the version and we don't want to call `info()` unless we know we can get the initialized status from it, even if calling it is supported.